server/log_rpc_server.go

```diff
@@ -573,6 +574,10 @@ func (t *TrillianLogRPCServer) GetLeavesByHash(ctx context.Context, req *trillia
 		return nil, err
 	}
 
+	if len(leaves) == 0 {
+		return nil, status.Errorf(codes.NotFound, "no leaves found with requested hashes")
```

Comment:
Won't this undo the skewness compensation stuff from #995 ?

IIRC/IIUC the idea was that in a slow-replicated setup, a `Get*` request might hit a lagging-behind server node.  That stale node might fail to find the requested information, but would include its known SLR in the response so that a client could detect this situation (and presumably retry in the hope of getting either a more up-to-date server node, or of time passing and the leaves appearing in the slow replica).

@AlCutter , @Martin2112 , am I remembering that right?

Reply:
Yes, it will prevent that working for this method. I'm not fully convinced we should change it. Might be a case to document the API better.